buildSrc/src/main/groovy/net/minecraftforge/forge/tasks/TeamcityRequests.groovy

```diff
@@ -0,0 +1,59 @@
+package net.minecraftforge.forge.tasks
+
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import groovy.transform.CompileStatic
+import org.eclipse.jgit.api.Git
+
+import javax.annotation.Nullable
+
+@CompileStatic
+class TeamcityRequests {
+    public static final Gson GSON = new Gson()
+
+    @Nullable
+    static <T> T jsonRequest(TypeToken<T> clazz, String url) throws Exception {
+        final HttpURLConnection conn = (HttpURLConnection) URI.create(url).toURL().openConnection()
+        conn.setRequestProperty('Accept', 'application/json')
+        conn.connect()
```

Comment:
Should we have read and connect timeouts configured for this connection, to ensure that we don't take too long with this task?